.github/workflows/main.yml

```diff
@@ -0,0 +1,56 @@
+name: main
+on:
+  pull_request:
+  push:
+    branches: [master, 'test-me-*']
```

Comment:
Probably don't want to keep `'test-me-*` in here

Reply:
oh, I usually find this useful to debug CI without having to open a PR

Reply:
Stupid question: If you can push to a repo and you trust others who can push to the repo, why bother restricting this at all?

Reply:
it's mostly so when pushing branches pushed to `pycqa/pycodestyle` and creating a pull request based on them you don't run the actions twice

Reply:
:shrug: You're still running them twice, just once on a random push, and a second time when opening the PR. They're just separate.

Reply:
in that case it would actually run 3 times:
- once when pushing
- once when opening the PR
- once when merged

the `branches: [...]` is to filter it down so it only runs twice:

- once when opening the PR
- once when merged

almost all of the CI systems get this wrong by default (imo) and spend twice as much time running tests than they need to be

Reply:
:shrug: I don't personally see any problems with running tests that often. But also CI systems never actually test against a merged build, just against the head of the branch so they're usually trash either way

Reply:
GA tests against the merged revision

Reply:
It doesn't update when master moves forward in my testing though so it's a point-in-time merged revision, not one before you press merge

Reply:
true true

Reply:
Not an argument against GA just a general gripe